// @maplibre/maplibre-react-native/lib/commonjs/MapLibreRN.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  UserTrackingModes: true,
  Camera: true,
  UserTrackingMode: true,
  MapView: true,
  Light: true,
  PointAnnotation: true,
  Annotation: true,
  Callout: true,
  requestAndroidLocationPermissions: true,
  UserLocation: true,
  UserLocationRenderMode: true,
  VectorSource: true,
  ShapeSource: true,
  RasterSource: true,
  ImageSource: true,
  Images: true,
  FillLayer: true,
  FillExtrusionLayer: true,
  HeatmapLayer: true,
  LineLayer: true,
  CircleLayer: true,
  SymbolLayer: true,
  RasterLayer: true,
  BackgroundLayer: true,
  locationManager: true,
  offlineManager: true,
  OfflinePack: true,
  OfflineCreatePackOptions: true,
  snapshotManager: true,
  MarkerView: true,
  Animated: true,
  Style: true,
  Logger: true
};
Object.defineProperty(exports, "Animated", {
  enumerable: true,
  get: function () {
    return _Animated.default;
  }
});
Object.defineProperty(exports, "Annotation", {
  enumerable: true,
  get: function () {
    return _Annotation.default;
  }
});
Object.defineProperty(exports, "BackgroundLayer", {
  enumerable: true,
  get: function () {
    return _BackgroundLayer.default;
  }
});
Object.defineProperty(exports, "Callout", {
  enumerable: true,
  get: function () {
    return _Callout.default;
  }
});
Object.defineProperty(exports, "Camera", {
  enumerable: true,
  get: function () {
    return _Camera.default;
  }
});
Object.defineProperty(exports, "CircleLayer", {
  enumerable: true,
  get: function () {
    return _CircleLayer.default;
  }
});
Object.defineProperty(exports, "FillExtrusionLayer", {
  enumerable: true,
  get: function () {
    return _FillExtrusionLayer.default;
  }
});
Object.defineProperty(exports, "FillLayer", {
  enumerable: true,
  get: function () {
    return _FillLayer.default;
  }
});
Object.defineProperty(exports, "HeatmapLayer", {
  enumerable: true,
  get: function () {
    return _HeatmapLayer.default;
  }
});
Object.defineProperty(exports, "ImageSource", {
  enumerable: true,
  get: function () {
    return _ImageSource.default;
  }
});
Object.defineProperty(exports, "Images", {
  enumerable: true,
  get: function () {
    return _Images.default;
  }
});
Object.defineProperty(exports, "Light", {
  enumerable: true,
  get: function () {
    return _Light.default;
  }
});
Object.defineProperty(exports, "LineLayer", {
  enumerable: true,
  get: function () {
    return _LineLayer.default;
  }
});
Object.defineProperty(exports, "Logger", {
  enumerable: true,
  get: function () {
    return _Logger.default;
  }
});
Object.defineProperty(exports, "MapView", {
  enumerable: true,
  get: function () {
    return _MapView.default;
  }
});
Object.defineProperty(exports, "MarkerView", {
  enumerable: true,
  get: function () {
    return _MarkerView.default;
  }
});
Object.defineProperty(exports, "OfflineCreatePackOptions", {
  enumerable: true,
  get: function () {
    return _OfflineCreatePackOptions.default;
  }
});
Object.defineProperty(exports, "OfflinePack", {
  enumerable: true,
  get: function () {
    return _OfflinePack.default;
  }
});
Object.defineProperty(exports, "PointAnnotation", {
  enumerable: true,
  get: function () {
    return _PointAnnotation.default;
  }
});
Object.defineProperty(exports, "RasterLayer", {
  enumerable: true,
  get: function () {
    return _RasterLayer.default;
  }
});
Object.defineProperty(exports, "RasterSource", {
  enumerable: true,
  get: function () {
    return _RasterSource.default;
  }
});
Object.defineProperty(exports, "ShapeSource", {
  enumerable: true,
  get: function () {
    return _ShapeSource.default;
  }
});
Object.defineProperty(exports, "Style", {
  enumerable: true,
  get: function () {
    return _Style.default;
  }
});
Object.defineProperty(exports, "SymbolLayer", {
  enumerable: true,
  get: function () {
    return _SymbolLayer.default;
  }
});
Object.defineProperty(exports, "UserLocation", {
  enumerable: true,
  get: function () {
    return _UserLocation.default;
  }
});
Object.defineProperty(exports, "UserLocationRenderMode", {
  enumerable: true,
  get: function () {
    return _UserLocation.UserLocationRenderMode;
  }
});
Object.defineProperty(exports, "UserTrackingMode", {
  enumerable: true,
  get: function () {
    return _Camera.UserTrackingMode;
  }
});
exports.UserTrackingModes = void 0;
Object.defineProperty(exports, "VectorSource", {
  enumerable: true,
  get: function () {
    return _VectorSource.default;
  }
});
Object.defineProperty(exports, "locationManager", {
  enumerable: true,
  get: function () {
    return _locationManager.default;
  }
});
Object.defineProperty(exports, "offlineManager", {
  enumerable: true,
  get: function () {
    return _offlineManager.default;
  }
});
Object.defineProperty(exports, "requestAndroidLocationPermissions", {
  enumerable: true,
  get: function () {
    return _requestAndroidLocationPermissions.requestAndroidLocationPermissions;
  }
});
Object.defineProperty(exports, "snapshotManager", {
  enumerable: true,
  get: function () {
    return _snapshotManager.default;
  }
});
var _Camera = _interopRequireWildcard(require("./components/Camera.js"));
var _MLNModule = require("./MLNModule.js");
Object.keys(_MLNModule).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _MLNModule[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _MLNModule[key];
    }
  });
});
var _MapView = _interopRequireDefault(require("./components/MapView.js"));
var _Light = _interopRequireDefault(require("./components/Light.js"));
var _PointAnnotation = _interopRequireDefault(require("./components/PointAnnotation.js"));
var _Annotation = _interopRequireDefault(require("./components/Annotation.js"));
var _Callout = _interopRequireDefault(require("./components/Callout.js"));
var _requestAndroidLocationPermissions = require("./requestAndroidLocationPermissions.js");
var _UserLocation = _interopRequireWildcard(require("./components/UserLocation.js"));
var _VectorSource = _interopRequireDefault(require("./components/VectorSource.js"));
var _ShapeSource = _interopRequireDefault(require("./components/ShapeSource.js"));
var _RasterSource = _interopRequireDefault(require("./components/RasterSource.js"));
var _ImageSource = _interopRequireDefault(require("./components/ImageSource.js"));
var _Images = _interopRequireDefault(require("./components/Images.js"));
var _FillLayer = _interopRequireDefault(require("./components/FillLayer.js"));
var _FillExtrusionLayer = _interopRequireDefault(require("./components/FillExtrusionLayer.js"));
var _HeatmapLayer = _interopRequireDefault(require("./components/HeatmapLayer.js"));
var _LineLayer = _interopRequireDefault(require("./components/LineLayer.js"));
var _CircleLayer = _interopRequireDefault(require("./components/CircleLayer.js"));
var _SymbolLayer = _interopRequireDefault(require("./components/SymbolLayer.js"));
var _RasterLayer = _interopRequireDefault(require("./components/RasterLayer.js"));
var _BackgroundLayer = _interopRequireDefault(require("./components/BackgroundLayer.js"));
var _locationManager = _interopRequireDefault(require("./modules/location/locationManager.js"));
var _offlineManager = _interopRequireDefault(require("./modules/offline/offlineManager.js"));
var _OfflinePack = _interopRequireDefault(require("./modules/offline/OfflinePack.js"));
var _OfflineCreatePackOptions = _interopRequireDefault(require("./modules/offline/OfflineCreatePackOptions.js"));
var _snapshotManager = _interopRequireDefault(require("./modules/snapshot/snapshotManager.js"));
var _MarkerView = _interopRequireDefault(require("./components/MarkerView.js"));
var _Animated = _interopRequireDefault(require("./utils/animated/Animated.js"));
var _Style = _interopRequireDefault(require("./components/Style.js"));
var _Logger = _interopRequireDefault(require("./utils/Logger.js"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/** @deprecated UserTrackingModes is deprecated use UserTrackingMode */
const UserTrackingModes = exports.UserTrackingModes = _Camera.UserTrackingMode;
