// @maplibre/maplibre-react-native/lib/commonjs/utils/MapLibreRNStyles.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VisibilityEnum = exports.TextWritingModeEnum = exports.TextVariableAnchorEnum = exports.TextTranslateAnchorEnum = exports.TextTransformEnum = exports.TextRotationAlignmentEnum = exports.TextPitchAlignmentEnum = exports.TextJustifyEnum = exports.TextAnchorEnum = exports.SymbolZOrderEnum = exports.SymbolPlacementEnum = exports.RasterResamplingEnum = exports.LineTranslateAnchorEnum = exports.LineJoinEnum = exports.LineCapEnum = exports.IconTranslateAnchorEnum = exports.IconTextFitEnum = exports.IconRotationAlignmentEnum = exports.IconPitchAlignmentEnum = exports.IconAnchorEnum = exports.HillshadeIlluminationAnchorEnum = exports.FillTranslateAnchorEnum = exports.FillExtrusionTranslateAnchorEnum = exports.CircleTranslateAnchorEnum = exports.CirclePitchScaleEnum = exports.CirclePitchAlignmentEnum = exports.AnchorEnum = void 0;
// DO NOT MODIFY
// This file is auto-generated from scripts/templates/MapLibreRNStyles.ts.ejs
/* TODO */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
let VisibilityEnum = exports.VisibilityEnum = /*#__PURE__*/function (VisibilityEnum) {
  VisibilityEnum["Visible"] = "visible";
  VisibilityEnum["None"] = "none";
  return VisibilityEnum;
}({});
let FillTranslateAnchorEnum = exports.FillTranslateAnchorEnum = /*#__PURE__*/function (FillTranslateAnchorEnum) {
  FillTranslateAnchorEnum["Map"] = "map";
  FillTranslateAnchorEnum["Viewport"] = "viewport";
  return FillTranslateAnchorEnum;
}({});
let LineCapEnum = exports.LineCapEnum = /*#__PURE__*/function (LineCapEnum) {
  LineCapEnum["Butt"] = "butt";
  LineCapEnum["Round"] = "round";
  LineCapEnum["Square"] = "square";
  return LineCapEnum;
}({});
let LineJoinEnum = exports.LineJoinEnum = /*#__PURE__*/function (LineJoinEnum) {
  LineJoinEnum["Bevel"] = "bevel";
  LineJoinEnum["Round"] = "round";
  LineJoinEnum["Miter"] = "miter";
  return LineJoinEnum;
}({});
let LineTranslateAnchorEnum = exports.LineTranslateAnchorEnum = /*#__PURE__*/function (LineTranslateAnchorEnum) {
  LineTranslateAnchorEnum["Map"] = "map";
  LineTranslateAnchorEnum["Viewport"] = "viewport";
  return LineTranslateAnchorEnum;
}({});
let SymbolPlacementEnum = exports.SymbolPlacementEnum = /*#__PURE__*/function (SymbolPlacementEnum) {
  SymbolPlacementEnum["Point"] = "point";
  SymbolPlacementEnum["Line"] = "line";
  SymbolPlacementEnum["LineCenter"] = "line-center";
  return SymbolPlacementEnum;
}({});
let SymbolZOrderEnum = exports.SymbolZOrderEnum = /*#__PURE__*/function (SymbolZOrderEnum) {
  SymbolZOrderEnum["Auto"] = "auto";
  SymbolZOrderEnum["ViewportY"] = "viewport-y";
  SymbolZOrderEnum["Source"] = "source";
  return SymbolZOrderEnum;
}({});
let IconRotationAlignmentEnum = exports.IconRotationAlignmentEnum = /*#__PURE__*/function (IconRotationAlignmentEnum) {
  IconRotationAlignmentEnum["Map"] = "map";
  IconRotationAlignmentEnum["Viewport"] = "viewport";
  IconRotationAlignmentEnum["Auto"] = "auto";
  return IconRotationAlignmentEnum;
}({});
let IconTextFitEnum = exports.IconTextFitEnum = /*#__PURE__*/function (IconTextFitEnum) {
  IconTextFitEnum["None"] = "none";
  IconTextFitEnum["Width"] = "width";
  IconTextFitEnum["Height"] = "height";
  IconTextFitEnum["Both"] = "both";
  return IconTextFitEnum;
}({});
let IconAnchorEnum = exports.IconAnchorEnum = /*#__PURE__*/function (IconAnchorEnum) {
  IconAnchorEnum["Center"] = "center";
  IconAnchorEnum["Left"] = "left";
  IconAnchorEnum["Right"] = "right";
  IconAnchorEnum["Top"] = "top";
  IconAnchorEnum["Bottom"] = "bottom";
  IconAnchorEnum["TopLeft"] = "top-left";
  IconAnchorEnum["TopRight"] = "top-right";
  IconAnchorEnum["BottomLeft"] = "bottom-left";
  IconAnchorEnum["BottomRight"] = "bottom-right";
  return IconAnchorEnum;
}({});
let IconPitchAlignmentEnum = exports.IconPitchAlignmentEnum = /*#__PURE__*/function (IconPitchAlignmentEnum) {
  IconPitchAlignmentEnum["Map"] = "map";
  IconPitchAlignmentEnum["Viewport"] = "viewport";
  IconPitchAlignmentEnum["Auto"] = "auto";
  return IconPitchAlignmentEnum;
}({});
let TextPitchAlignmentEnum = exports.TextPitchAlignmentEnum = /*#__PURE__*/function (TextPitchAlignmentEnum) {
  TextPitchAlignmentEnum["Map"] = "map";
  TextPitchAlignmentEnum["Viewport"] = "viewport";
  TextPitchAlignmentEnum["Auto"] = "auto";
  return TextPitchAlignmentEnum;
}({});
let TextRotationAlignmentEnum = exports.TextRotationAlignmentEnum = /*#__PURE__*/function (TextRotationAlignmentEnum) {
  TextRotationAlignmentEnum["Map"] = "map";
  TextRotationAlignmentEnum["Viewport"] = "viewport";
  TextRotationAlignmentEnum["ViewportGlyph"] = "viewport-glyph";
  TextRotationAlignmentEnum["Auto"] = "auto";
  return TextRotationAlignmentEnum;
}({});
let TextJustifyEnum = exports.TextJustifyEnum = /*#__PURE__*/function (TextJustifyEnum) {
  TextJustifyEnum["Auto"] = "auto";
  TextJustifyEnum["Left"] = "left";
  TextJustifyEnum["Center"] = "center";
  TextJustifyEnum["Right"] = "right";
  return TextJustifyEnum;
}({});
let TextVariableAnchorEnum = exports.TextVariableAnchorEnum = /*#__PURE__*/function (TextVariableAnchorEnum) {
  TextVariableAnchorEnum["Center"] = "center";
  TextVariableAnchorEnum["Left"] = "left";
  TextVariableAnchorEnum["Right"] = "right";
  TextVariableAnchorEnum["Top"] = "top";
  TextVariableAnchorEnum["Bottom"] = "bottom";
  TextVariableAnchorEnum["TopLeft"] = "top-left";
  TextVariableAnchorEnum["TopRight"] = "top-right";
  TextVariableAnchorEnum["BottomLeft"] = "bottom-left";
  TextVariableAnchorEnum["BottomRight"] = "bottom-right";
  return TextVariableAnchorEnum;
}({});
let TextAnchorEnum = exports.TextAnchorEnum = /*#__PURE__*/function (TextAnchorEnum) {
  TextAnchorEnum["Center"] = "center";
  TextAnchorEnum["Left"] = "left";
  TextAnchorEnum["Right"] = "right";
  TextAnchorEnum["Top"] = "top";
  TextAnchorEnum["Bottom"] = "bottom";
  TextAnchorEnum["TopLeft"] = "top-left";
  TextAnchorEnum["TopRight"] = "top-right";
  TextAnchorEnum["BottomLeft"] = "bottom-left";
  TextAnchorEnum["BottomRight"] = "bottom-right";
  return TextAnchorEnum;
}({});
let TextWritingModeEnum = exports.TextWritingModeEnum = /*#__PURE__*/function (TextWritingModeEnum) {
  TextWritingModeEnum["Horizontal"] = "horizontal";
  TextWritingModeEnum["Vertical"] = "vertical";
  return TextWritingModeEnum;
}({});
let TextTransformEnum = exports.TextTransformEnum = /*#__PURE__*/function (TextTransformEnum) {
  TextTransformEnum["None"] = "none";
  TextTransformEnum["Uppercase"] = "uppercase";
  TextTransformEnum["Lowercase"] = "lowercase";
  return TextTransformEnum;
}({});
let IconTranslateAnchorEnum = exports.IconTranslateAnchorEnum = /*#__PURE__*/function (IconTranslateAnchorEnum) {
  IconTranslateAnchorEnum["Map"] = "map";
  IconTranslateAnchorEnum["Viewport"] = "viewport";
  return IconTranslateAnchorEnum;
}({});
let TextTranslateAnchorEnum = exports.TextTranslateAnchorEnum = /*#__PURE__*/function (TextTranslateAnchorEnum) {
  TextTranslateAnchorEnum["Map"] = "map";
  TextTranslateAnchorEnum["Viewport"] = "viewport";
  return TextTranslateAnchorEnum;
}({});
let CircleTranslateAnchorEnum = exports.CircleTranslateAnchorEnum = /*#__PURE__*/function (CircleTranslateAnchorEnum) {
  CircleTranslateAnchorEnum["Map"] = "map";
  CircleTranslateAnchorEnum["Viewport"] = "viewport";
  return CircleTranslateAnchorEnum;
}({});
let CirclePitchScaleEnum = exports.CirclePitchScaleEnum = /*#__PURE__*/function (CirclePitchScaleEnum) {
  CirclePitchScaleEnum["Map"] = "map";
  CirclePitchScaleEnum["Viewport"] = "viewport";
  return CirclePitchScaleEnum;
}({});
let CirclePitchAlignmentEnum = exports.CirclePitchAlignmentEnum = /*#__PURE__*/function (CirclePitchAlignmentEnum) {
  CirclePitchAlignmentEnum["Map"] = "map";
  CirclePitchAlignmentEnum["Viewport"] = "viewport";
  return CirclePitchAlignmentEnum;
}({});
let FillExtrusionTranslateAnchorEnum = exports.FillExtrusionTranslateAnchorEnum = /*#__PURE__*/function (FillExtrusionTranslateAnchorEnum) {
  FillExtrusionTranslateAnchorEnum["Map"] = "map";
  FillExtrusionTranslateAnchorEnum["Viewport"] = "viewport";
  return FillExtrusionTranslateAnchorEnum;
}({});
let RasterResamplingEnum = exports.RasterResamplingEnum = /*#__PURE__*/function (RasterResamplingEnum) {
  RasterResamplingEnum["Linear"] = "linear";
  RasterResamplingEnum["Nearest"] = "nearest";
  return RasterResamplingEnum;
}({});
let HillshadeIlluminationAnchorEnum = exports.HillshadeIlluminationAnchorEnum = /*#__PURE__*/function (HillshadeIlluminationAnchorEnum) {
  HillshadeIlluminationAnchorEnum["Map"] = "map";
  HillshadeIlluminationAnchorEnum["Viewport"] = "viewport";
  return HillshadeIlluminationAnchorEnum;
}({});
let AnchorEnum = exports.AnchorEnum = /*#__PURE__*/function (AnchorEnum) {
  AnchorEnum["Map"] = "map";
  AnchorEnum["Viewport"] = "viewport";
  return AnchorEnum;
}({});
