// @maplibre/maplibre-react-native/lib/module/utils/MapLibreRNStyles.d.js
"use strict";

// DO NOT MODIFY
// This file is auto-generated from scripts/templates/MapLibreRNStyles.ts.ejs

/* TODO */

// eslint-disable-next-line @typescript-eslint/no-unused-vars

export let VisibilityEnum = /*#__PURE__*/function (VisibilityEnum) {
  VisibilityEnum["Visible"] = "visible";
  VisibilityEnum["None"] = "none";
  return VisibilityEnum;
}({});
export let FillTranslateAnchorEnum = /*#__PURE__*/function (FillTranslateAnchorEnum) {
  FillTranslateAnchorEnum["Map"] = "map";
  FillTranslateAnchorEnum["Viewport"] = "viewport";
  return FillTranslateAnchorEnum;
}({});
export let LineCapEnum = /*#__PURE__*/function (LineCapEnum) {
  LineCapEnum["Butt"] = "butt";
  LineCapEnum["Round"] = "round";
  LineCapEnum["Square"] = "square";
  return LineCapEnum;
}({});
export let LineJoinEnum = /*#__PURE__*/function (LineJoinEnum) {
  LineJoinEnum["Bevel"] = "bevel";
  LineJoinEnum["Round"] = "round";
  LineJoinEnum["Miter"] = "miter";
  return LineJoinEnum;
}({});
export let LineTranslateAnchorEnum = /*#__PURE__*/function (LineTranslateAnchorEnum) {
  LineTranslateAnchorEnum["Map"] = "map";
  LineTranslateAnchorEnum["Viewport"] = "viewport";
  return LineTranslateAnchorEnum;
}({});
export let SymbolPlacementEnum = /*#__PURE__*/function (SymbolPlacementEnum) {
  SymbolPlacementEnum["Point"] = "point";
  SymbolPlacementEnum["Line"] = "line";
  SymbolPlacementEnum["LineCenter"] = "line-center";
  return SymbolPlacementEnum;
}({});
export let SymbolZOrderEnum = /*#__PURE__*/function (SymbolZOrderEnum) {
  SymbolZOrderEnum["Auto"] = "auto";
  SymbolZOrderEnum["ViewportY"] = "viewport-y";
  SymbolZOrderEnum["Source"] = "source";
  return SymbolZOrderEnum;
}({});
export let IconRotationAlignmentEnum = /*#__PURE__*/function (IconRotationAlignmentEnum) {
  IconRotationAlignmentEnum["Map"] = "map";
  IconRotationAlignmentEnum["Viewport"] = "viewport";
  IconRotationAlignmentEnum["Auto"] = "auto";
  return IconRotationAlignmentEnum;
}({});
export let IconTextFitEnum = /*#__PURE__*/function (IconTextFitEnum) {
  IconTextFitEnum["None"] = "none";
  IconTextFitEnum["Width"] = "width";
  IconTextFitEnum["Height"] = "height";
  IconTextFitEnum["Both"] = "both";
  return IconTextFitEnum;
}({});
export let IconAnchorEnum = /*#__PURE__*/function (IconAnchorEnum) {
  IconAnchorEnum["Center"] = "center";
  IconAnchorEnum["Left"] = "left";
  IconAnchorEnum["Right"] = "right";
  IconAnchorEnum["Top"] = "top";
  IconAnchorEnum["Bottom"] = "bottom";
  IconAnchorEnum["TopLeft"] = "top-left";
  IconAnchorEnum["TopRight"] = "top-right";
  IconAnchorEnum["BottomLeft"] = "bottom-left";
  IconAnchorEnum["BottomRight"] = "bottom-right";
  return IconAnchorEnum;
}({});
export let IconPitchAlignmentEnum = /*#__PURE__*/function (IconPitchAlignmentEnum) {
  IconPitchAlignmentEnum["Map"] = "map";
  IconPitchAlignmentEnum["Viewport"] = "viewport";
  IconPitchAlignmentEnum["Auto"] = "auto";
  return IconPitchAlignmentEnum;
}({});
export let TextPitchAlignmentEnum = /*#__PURE__*/function (TextPitchAlignmentEnum) {
  TextPitchAlignmentEnum["Map"] = "map";
  TextPitchAlignmentEnum["Viewport"] = "viewport";
  TextPitchAlignmentEnum["Auto"] = "auto";
  return TextPitchAlignmentEnum;
}({});
export let TextRotationAlignmentEnum = /*#__PURE__*/function (TextRotationAlignmentEnum) {
  TextRotationAlignmentEnum["Map"] = "map";
  TextRotationAlignmentEnum["Viewport"] = "viewport";
  TextRotationAlignmentEnum["ViewportGlyph"] = "viewport-glyph";
  TextRotationAlignmentEnum["Auto"] = "auto";
  return TextRotationAlignmentEnum;
}({});
export let TextJustifyEnum = /*#__PURE__*/function (TextJustifyEnum) {
  TextJustifyEnum["Auto"] = "auto";
  TextJustifyEnum["Left"] = "left";
  TextJustifyEnum["Center"] = "center";
  TextJustifyEnum["Right"] = "right";
  return TextJustifyEnum;
}({});
export let TextVariableAnchorEnum = /*#__PURE__*/function (TextVariableAnchorEnum) {
  TextVariableAnchorEnum["Center"] = "center";
  TextVariableAnchorEnum["Left"] = "left";
  TextVariableAnchorEnum["Right"] = "right";
  TextVariableAnchorEnum["Top"] = "top";
  TextVariableAnchorEnum["Bottom"] = "bottom";
  TextVariableAnchorEnum["TopLeft"] = "top-left";
  TextVariableAnchorEnum["TopRight"] = "top-right";
  TextVariableAnchorEnum["BottomLeft"] = "bottom-left";
  TextVariableAnchorEnum["BottomRight"] = "bottom-right";
  return TextVariableAnchorEnum;
}({});
export let TextAnchorEnum = /*#__PURE__*/function (TextAnchorEnum) {
  TextAnchorEnum["Center"] = "center";
  TextAnchorEnum["Left"] = "left";
  TextAnchorEnum["Right"] = "right";
  TextAnchorEnum["Top"] = "top";
  TextAnchorEnum["Bottom"] = "bottom";
  TextAnchorEnum["TopLeft"] = "top-left";
  TextAnchorEnum["TopRight"] = "top-right";
  TextAnchorEnum["BottomLeft"] = "bottom-left";
  TextAnchorEnum["BottomRight"] = "bottom-right";
  return TextAnchorEnum;
}({});
export let TextWritingModeEnum = /*#__PURE__*/function (TextWritingModeEnum) {
  TextWritingModeEnum["Horizontal"] = "horizontal";
  TextWritingModeEnum["Vertical"] = "vertical";
  return TextWritingModeEnum;
}({});
export let TextTransformEnum = /*#__PURE__*/function (TextTransformEnum) {
  TextTransformEnum["None"] = "none";
  TextTransformEnum["Uppercase"] = "uppercase";
  TextTransformEnum["Lowercase"] = "lowercase";
  return TextTransformEnum;
}({});
export let IconTranslateAnchorEnum = /*#__PURE__*/function (IconTranslateAnchorEnum) {
  IconTranslateAnchorEnum["Map"] = "map";
  IconTranslateAnchorEnum["Viewport"] = "viewport";
  return IconTranslateAnchorEnum;
}({});
export let TextTranslateAnchorEnum = /*#__PURE__*/function (TextTranslateAnchorEnum) {
  TextTranslateAnchorEnum["Map"] = "map";
  TextTranslateAnchorEnum["Viewport"] = "viewport";
  return TextTranslateAnchorEnum;
}({});
export let CircleTranslateAnchorEnum = /*#__PURE__*/function (CircleTranslateAnchorEnum) {
  CircleTranslateAnchorEnum["Map"] = "map";
  CircleTranslateAnchorEnum["Viewport"] = "viewport";
  return CircleTranslateAnchorEnum;
}({});
export let CirclePitchScaleEnum = /*#__PURE__*/function (CirclePitchScaleEnum) {
  CirclePitchScaleEnum["Map"] = "map";
  CirclePitchScaleEnum["Viewport"] = "viewport";
  return CirclePitchScaleEnum;
}({});
export let CirclePitchAlignmentEnum = /*#__PURE__*/function (CirclePitchAlignmentEnum) {
  CirclePitchAlignmentEnum["Map"] = "map";
  CirclePitchAlignmentEnum["Viewport"] = "viewport";
  return CirclePitchAlignmentEnum;
}({});
export let FillExtrusionTranslateAnchorEnum = /*#__PURE__*/function (FillExtrusionTranslateAnchorEnum) {
  FillExtrusionTranslateAnchorEnum["Map"] = "map";
  FillExtrusionTranslateAnchorEnum["Viewport"] = "viewport";
  return FillExtrusionTranslateAnchorEnum;
}({});
export let RasterResamplingEnum = /*#__PURE__*/function (RasterResamplingEnum) {
  RasterResamplingEnum["Linear"] = "linear";
  RasterResamplingEnum["Nearest"] = "nearest";
  return RasterResamplingEnum;
}({});
export let HillshadeIlluminationAnchorEnum = /*#__PURE__*/function (HillshadeIlluminationAnchorEnum) {
  HillshadeIlluminationAnchorEnum["Map"] = "map";
  HillshadeIlluminationAnchorEnum["Viewport"] = "viewport";
  return HillshadeIlluminationAnchorEnum;
}({});
export let AnchorEnum = /*#__PURE__*/function (AnchorEnum) {
  AnchorEnum["Map"] = "map";
  AnchorEnum["Viewport"] = "viewport";
  return AnchorEnum;
}({});
