// @maplibre/maplibre-react-native/lib/commonjs/utils/animated/Animated.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.AnimatedShapeSource = void 0;
var _reactNative = require("react-native");
var _AnimatedCoordinatesArray = _interopRequireDefault(require("./AnimatedCoordinatesArray.js"));
var _AnimatedExtractCoordinateFromArray = _interopRequireDefault(require("./AnimatedExtractCoordinateFromArray.js"));
var _AnimatedRouteCoordinatesArray = _interopRequireDefault(require("./AnimatedRouteCoordinatesArray.js"));
var _AnimatedShape = _interopRequireDefault(require("./AnimatedShape.js"));
var _BackgroundLayer = _interopRequireDefault(require("../../components/BackgroundLayer.js"));
var _CircleLayer = _interopRequireDefault(require("../../components/CircleLayer.js"));
var _FillExtrusionLayer = _interopRequireDefault(require("../../components/FillExtrusionLayer.js"));
var _FillLayer = _interopRequireDefault(require("../../components/FillLayer.js"));
var _ImageSource = _interopRequireDefault(require("../../components/ImageSource.js"));
var _LineLayer = _interopRequireDefault(require("../../components/LineLayer.js"));
var _RasterLayer = _interopRequireDefault(require("../../components/RasterLayer.js"));
var _ShapeSource = _interopRequireDefault(require("../../components/ShapeSource.js"));
var _SymbolLayer = _interopRequireDefault(require("../../components/SymbolLayer.js"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const Animated = {
  // sources
  ShapeSource: _reactNative.Animated.createAnimatedComponent(_ShapeSource.default),
  ImageSource: _reactNative.Animated.createAnimatedComponent(_ImageSource.default),
  // layers
  FillLayer: _reactNative.Animated.createAnimatedComponent(_FillLayer.default),
  FillExtrusionLayer: _reactNative.Animated.createAnimatedComponent(_FillExtrusionLayer.default),
  LineLayer: _reactNative.Animated.createAnimatedComponent(_LineLayer.default),
  CircleLayer: _reactNative.Animated.createAnimatedComponent(_CircleLayer.default),
  SymbolLayer: _reactNative.Animated.createAnimatedComponent(_SymbolLayer.default),
  RasterLayer: _reactNative.Animated.createAnimatedComponent(_RasterLayer.default),
  BackgroundLayer: _reactNative.Animated.createAnimatedComponent(_BackgroundLayer.default),
  // values
  CoordinatesArray: _AnimatedCoordinatesArray.default,
  RouteCoordinatesArray: _AnimatedRouteCoordinatesArray.default,
  Shape: _AnimatedShape.default,
  ExtractCoordinateFromArray: _AnimatedExtractCoordinateFromArray.default
};
/**
 * Manual typing is required for AnimatedShapeSource because the
 * following error:
 * `Type instantiation is excessively deep and possibly infinite.ts(2589)`
 */
const AnimatedShapeSource = exports.AnimatedShapeSource = _reactNative.Animated.createAnimatedComponent(_ShapeSource.default);
var _default = exports.default = Animated;
